File: 2 CPP/Codeforces-540A-CombinationLock.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int n;
    cin >> n;
    string original, target;
    cin >> original >> target;
    int moves = 0;
    for(int i=0;i<n;++i){
        int o = original[i]-'0';
        int t = target[i]-'0';
        moves += min(abs(o-t),10-abs(o-t));
    }
    cout << moves << "\n";
}

File: 2 CPP/Codeforces-540A-CombinationLock.go
package main
import "fmt"

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}

func main(){
    var n int
    fmt.Scan(&n)
    var original string
    fmt.Scan(&original)
    var target string
    fmt.Scan(&target)
    var ans int = 0 
    for i:=0;i<n;i++ {
        var ato, att int
        ato = int((original[i])-'0')
        att = int((target[i])-'0')
        var diff int = abs(ato-att)
        ans += min(diff,10-diff)
    }
    fmt.Println(ans)
}
